frontend/src/services/UserFunctions.ts

```diff
@@ -36,3 +37,19 @@ export async function fetchAdminStatus(): Promise<SuccessObject> {
         });
     });
 }
+
+export const doesUserExist = async (username: string) => {
+    try {
+      const result = await callUserFunction(`check-username?username=${encodeURIComponent(username)}`);
```

Comment:
1. avoid magic string
2. why use `encodeURIComponent()` ? iirc its for ensuring that special characters are converted into a format that can be safely included in a URL, but theres alr a regex check

Reply:
correct me if im wrong but if ure alr allowing only alphanumeric to pass through (regex) as a 1st filter, wont the 2nd filter `encodeURIComponent` be redundant ?